process/block/preprocess/transactions.go

```diff
@@ -1071,12 +1080,12 @@ func (txs *transactions) ProcessMiniBlock(
 	for index := range miniBlockTxs {
 		if !haveTime() {
 			err = process.ErrTimeIsOut
-			return err
+			return processedTxHashes, err
 		}
 
 		err = txs.txProcessor.ProcessTransaction(miniBlockTxs[index])
 		if err != nil {
-			return err
+			return processedTxHashes, err
```

Comment:
I think here the situation is different. You should create another list for your scope, as in this point all processedTxHashes are already added but not yet processed. This list is used for revert gasRefunded and gasConsumed in case of an error raised in in this for. So no matter in which point the ProcessTransaction will fail you will return the same and actually the whole list always. If this is ok also for your case you can let as it is.

Reply:
this is correct. as reverting will happen only for those which were processed. if there is not match of the hash in the saved map than nothing happens.

Reply:
ok